### PR TITLE
fix(web): redaction edge-cases + multi-line array secret redaction in raw config view (#367)

### LINKED ISSUE #367: Settings raw config view: redaction edge-cases + buildRawFileTOML error disambiguation (follow-up from #363)
## Summary

Non-blocking follow-ups surfaced during the #363 (settings UI polish) Codoki review of the raw config-file view + redaction. The blocking High (brittle `redactRawTOML` that could leak secrets for no-space `key=value` / `[[array-of-tables]]`) was fixed and merged in #363 (`84934c5`). These three are hardening/clarity items deferred out of #363 to keep it scoped — none is a live secret-leak.

## Items

1. **`redactRawTOML` drops trailing inline comments on sensitive lines.** When a sensitive `key = value` line carries a trailing `# comment`, the redaction path may drop/alter the trailing comment rather than preserving it verbatim. The raw view aims to mirror the file faithfully (minus the redacted value), so a trailing comment on a sensitive line should survive.

2. **`buildRawFileTOML` returns `""` for two distinct cases.** It returns an empty string for both "config file not present / unconfigured" AND a read-error. The caller can't distinguish "nothing to show" from "failed to read the file" — they should be disambiguated (e.g. a sentinel/error return) so a genuine read failure isn't silently rendered as an empty raw view.

3. **More TOML edge-case tests for the raw redaction.** `TestRedactRawTOML` covers the four cases fixed in #363 (no-space, `[[section]]`, comment preservation, normal). Add coverage for further edge cases: trailing inline comments on sensitive lines (item 1), quoted keys, keys with dotted/sub-table paths, multi-line array values, and whitespace/indentation variants.

## Context
- Source: Codoki review on PR #363 (findings F2/F3/F4), triaged non-blocking.
- Surface: `internal/web/settings.go` (`redactRawTOML`, `buildRawFileTOML`) + its `_test.go`.

## Acceptance
- Trailing inline comments on sensitive lines are preserved (value redacted, comment intact).
- `buildRawFileTOML` distinguishes unconfigured from read-error.
- Added edge-case tests pass; no regression to the existing redaction behavior.

### PR DESCRIPTION
**Pull request by sydlexius**
## What

Harden secret redaction in the settings raw config-file view + disambiguate read errors. Closes #367 (follow-up from #363).

## Changes (`internal/web/settings.go`)

- **Trailing comments preserved on sensitive lines.** A sensitive `key = value # comment` now redacts only the value and keeps the trailing comment verbatim, via a quote-aware `trailingTOMLComment` (a `#` inside a `"basic"` or `'literal'` string is value, not a comment; handles `\"` escapes).
- **Multi-line sensitive arrays fully redacted.** A sensitive value spanning lines (e.g. `webhook_api_keys = [` then elements then `]`) previously redacted only the first line and emitted the continuation elements **verbatim** — a real secret leak in the read-only raw view. It now collapses to a single `key = "(redacted)"` line; continuation lines are consumed via a quote-aware `tomlBracketDepth` (a `]`/`[` inside a string doesn't count). Non-sensitive multi-line arrays still render verbatim.
- **All sensitive key forms matched** (quoted keys, dotted/sub-table paths, `[[array-of-tables]]`, no-space `key=value`) via `normalizeTOMLKeyPath`; #363's behavior preserved.
- **`buildRawFileTOML` returns `(string, error)`** — the template renders a read-error state distinctly from unconfigured; the path-bearing error goes only to `slog`, never the UI.

## Test plan
- `TestRedactRawTOMLEdgeCases` (incl. the multi-line `webhook_api_keys` repro asserting neither secret appears, and a `]`-inside-quoted-element case), `TestTrailingTOMLComment`, `TestBuildRawFileTOML`, `TestSettingsRawFileReadErrorRendered`. Every redaction case asserts the secret string never appears in output.
- `make gate` green; `internal/web/settings.go` patch coverage 100%.
- A pre-push adversarial review caught the multi-line-array HIGH leak; this PR fixes it (the redaction was single-line-only).

Sensitivity sourced from the existing `config.Registry()` `spec.Sensitive` set — no new notion of sensitivity.

## Release Notes

* **Bug Fixes**
  * Settings page now clearly distinguishes between a missing configuration file and file read errors, displaying specific error messages when access issues occur.
  * Enhanced sensitive data redaction in configuration file display to safely handle edge cases while preserving formatting and inline comments.

### CODE REVIEW

File: internal/web/settings_test.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package web
 import (
 	"net/http"
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/web/settings_test.go:4]: False positive (stale). There is no `fmt` import in this file on the current head (b2370ae): `grep '"fmt"' internal/web/settings_test.go` returns nothing and there are 0 `fmt.` usages. `go vet ./internal/web/` is clean and all CI is green (Test/Lint/Analyze Go/Build x5 = SUCCESS) — Go would not compile with an unused import. This is carried from an intermediate review state; no change needed.

**Comment by sydlexius** [comment]
@coderabbitai full review

**Comment by sydlexius** [comment]
@codoki Both High findings on this PR are false positives:

1. **"Unused `fmt` import"** — there is no `fmt` import in `internal/web/settings_test.go` on the current head (b2370ae): grep for the import returns nothing, 0 `fmt.` usages, `go vet ./internal/web/` is clean, and all CI is green (Test/Lint/Analyze Go/Build x5 = SUCCESS). Go would not compile with an unused import — this is stale from an intermediate review state.

2. **"Potential secret exposure" (settings_test.go:417/422/430)** — these are test fixtures (`super-secret-1`/`super-secret-2`/`super-secret-token`) that drive the redaction tests; the tests assert these values are NEVER present in the redacted output. They are not real credentials — they are precisely the surface this PR hardens.

All CI green, 0 unresolved threads. Please re-review.

File: internal/web/settings.go [hunk 2]
```diff
@@ -639,20 +656,48 @@ func redactRawTOML(raw string) string {
...
 		// all surrounding brackets yields the dotted-section name in either case.
+		// normalizeTOMLKeyPath unquotes any quoted segments so a header written as
+		// ["server"."tls"] resolves to the same dotted name as [server.tls].
 		if strings.HasPrefix(trimmed, "[") && strings.HasSuffix(trimmed, "]") {
-			section = strings.Trim(trimmed, "[]")
+			section = normalizeTOMLKeyPath(strings.Trim(trimmed, "[]"))
 			b.WriteString(line)
 			b.WriteByte('\n')
 			continue
 		}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/web/settings.go:677]: Fixed in 8a40af1. Table headers now strip a trailing inline comment (quote-aware, via trailingTOMLComment) before the [...] header test, so `[server] # note` still updates the current section and a following sensitive key is matched and redacted. Added a repro test.

File: internal/web/settings.go [hunk 3]
```diff
@@ -639,20 +656,48 @@ func redactRawTOML(raw string) string {
...
+				// The placeholder above already stands in for the whole value, so
+				// drop every continuation line until the array's brackets balance
+				// (the closing ']' line is part of the value and is dropped too).
+				// Without this, a multi-line array of secrets would leak every
+				// element after the first verbatim.
+				depth := tomlBracketDepth(line[eq+1:], 0)
+				for depth > 0 && i+1 < len(lines) {
+					i++
+					depth = tomlBracketDepth(lines[i], depth)
+				}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/web/settings.go:712]: Fixed in 8a40af1. A sensitive value that opens an unterminated multi-line basic (""") or literal (''') string now drops continuation lines until the matching closing delimiter (new openMultilineTOMLString helper), collapsing to a single redacted line; a same-line-closed triple-quote is not over-consumed. Added basic/literal/single-line repro tests.
